lib/src/util/condition_scope_visitor.dart

```diff
@@ -255,6 +267,20 @@ abstract class ConditionScopeVisitor extends RecursiveAstVisitor {
   }
 
   @override
+  visitSwitchCase(SwitchCase node) {
+    node.visitChildren(this);
+    AstNode previousCase = node.parent.childEntities
+        .where((e) => e is AstNode)
+        .lastWhere((n) => n != node, orElse: () => null);
+    if (previousCase == null ||
+        previousCase.childEntities.last is! BreakStatement) {
```

Comment:
I think this also needs to account for return and throw statements, and all or some continue statements.